src/software/thunderscope/gl/graphics/gl_polygon.py

```
@@ -62,7 +66,10 @@ def _update_shape_data(self) -> None:
         if not self.points:
             return
```

What are your thoughts on logging something here as an error/warning? I think this error could easily be buried otherwise if we just silently return early.